.github/workflows/cleanup-terraform.yaml

```diff
@@ -0,0 +1,33 @@
+# Workflow that runs whenever a PR is closed or merged and deletes any
+# AWS resources created by the `deploy` workflow.
+name: cleanup-terraform
+
+on:
+  pull_request:
+    types: [closed]
```

Comment:
As an aside on this workflow, we should also have a separate workflow to delete erroneous or useless runs. I broke this out into a separate issue in #56.